tests/tests.js

```diff
@@ -1957,4 +1957,244 @@ $(function() {
+        user1_session_token = data.session_token
+        ok(true, 'Logged in as user1');
+      }).on('error', function() {
+        ok(false, 'Failed to login as user1');
+      }).on('complete', createACL);
+    }
+
+    function createACL() {            
+      msg = "Create ACL as user1";
+      acl.members = [user2_id]
+      webservice.updateACL(acl).on('error', function() {
+        ok(false, msg);
+      }).on('success', function(data) {
+        acl_id = Object.keys(data)[0]
+        ok(true, msg);
+      }).on('complete', updateObjectWithACL);
+    }
+
+    function updateObjectWithACL() {
+      webservice.update(user_object_key, {__access__:acl_id}).on('success', function(data) {
+        ok(true, 'User1 updated the test object');
+      }).on('error', function() {
+        ok(false, 'User1 failed to update the test object');
+      }).on('complete', logoutUser1_CreateACL);
+    }
+
+    function logoutUser1_CreateACL() {
+      webservice.logout({email: user1.email, password: user1.password, session_token: user1_session_token}).on('success', function(data) {
+        ok(true, 'User1 logged out');
+      }).on('error', function() {
+        ok(false, 'Failed to logout as user1');
+      }).on('complete', loginUser2_ACL);
+    }
+
+    function loginUser2_ACL() {
+      webservice.login(user2).on('success', function(data) {
+        user2_session_token = data.session_token
+        ok(true, 'Logged in as user2');
+      }).on('error', function() {
+        ok(false, 'Failed to login as user2');
+      }).on('complete',  verifyACLCreated);
+    }
+
+    function verifyACLCreated() {
+      msg = "User2 got user1 data";
+      webservice.get(user_object_key, {applevel: false}).on('success', function() {
+        ok(true, msg)
+      }).on('error', function() {
+        ok(false, 'Could not find user1 data');
+      }).on('complete', logoutUser2_ACL);
+    }
+    
+    function logoutUser2_ACL() {
+      webservice.logout({email: user2.email, password: user2.password, session_token: user2_session_token}).on('success', function(data) {
+        ok(true, 'User2 logged out');
+      }).on('error', function() {
+        ok(false, 'Failed to logout as user2.');
+      }).on('complete', loginUser1_DeleteACL);
+    }
+
+    function loginUser1_DeleteACL() {
+      webservice.login(user1).on('success', function(data) {
+        user1_session_token = data.session_token
+        ok(true, 'Logged in as user1');
+      }).on('error', function() {
+        ok(false, 'Failed to login as user1');
+      }).on('complete', deleteACL);
+    }
+
+    function deleteACL() {
+      var msg = 'Test user ACL was deleted.';
+      webservice.deleteACL(acl_id).on('success', function() {
+        ok(true, msg);
+      }).on('error', function() {
+        ok(false, msg);
+      }).on('complete', logoutUser1_DeleteACL);
+    }
+
+    function logoutUser1_DeleteACL() {
+      webservice.login(user1).on('success', function(data) {
```

Comment:
You intended to logout here but you logged in. Is all the logging out and back in necessary? Can you stay logged in?